### ISSUE TITLE
Debugger fails start

### ISSUE THREAD

**Issue by batchu** [microsoft/vscode-react-native#1067]
## Actual Behavior
1. Debugger fails to start

## Expected Behavior
1. After creating a debug config, debugger should start as usual
2. Be able to put breakpoints

## Software versions
- React-Native VS Code extension version: 
- VSCode version: 0.10.2
- OS platform and version:  Windows 10.0.17763 N/A Build 17763
- NodeJS version: 10.16.0
- React Native version:     "expo": "^33.0.0",
    "react": "16.8.3",
    "react-dom": "^16.8.6",
    "react-native": "https://github.com/expo/react-native/archive/sdk-33.0.0.tar.gz",
    "react-native-web": "^0.11.4"
- Expo SDK version (include if relevant): 33.0.0

## Outputs (Include if relevant)
[Output ](https://ibb.co/y8gy2X2)

**Comment by SounD120** [comment]
Hi @batchu and thanks for reaching us.
For debugging Expo application there is `Debug in Exponent` debug scenario instead of `Debug Android`.  This scenario will run Expo tunnel, so you don't need to execute `expo start` command. We are also have an [instruction for Expo debugging](https://github.com/microsoft/vscode-react-native/blob/master/doc/expo.md). I'd recommend to ensure that Expo server is not running and then to debug your application using the instruction above. Please let us know if you have any questions.

**Comment by batchu** [comment]
@SounD120 I did go through that tutorial and I experienced the same behavior so I switched to "Debug in Android"

Here's what I see when I try the Expo method
https://ibb.co/L1sFz8Z

Thank you.

**Comment by SounD120** [comment]
@batchu could you please attach your outputs by the format pointed below?
- Output of the Debug Console (View -> Toggle Debug Console):
```
PASTE OUTPUT HERE
```
- Output of the React-Native output channel (View -> Toggle Output -> Select React-Native in ListBox):
```
PASTE OUTPUT HERE
```
- Output of the Developer Tools console (Help -> Toggle Developer Tools -> Select Console tab):
```
PASTE OUTPUT HERE
```

**Comment by cvharris** [comment]
I'm seeing the same thing. Nothing in any of the three outputs, just click 'Run' and it thinks for a sec before silently stopping. Maybe something to do with latest VSCode?

**Comment by batchu** [comment]
@SounD120 I second @cvharris. I do not see any output as well.

**Comment by Saraceni** [comment]
Having the same problem but I'm not using expo. I click play to start the Attach to packager config, it tries to runs but stops without any output

**Comment by cvharris** [comment]
I'm not using expo either.

It appears to be an issue on startup. If I restart the window (Cmd + P -> Reload Window) then debugger starts again.

Maybe the extension is failing to do something if previous Code windows are reopened on computer restart? This was an issue for me after upgrading Mac which required restart, and just now when my battery died again.

**Comment by ck86** [comment]
It's not starting anymore with the latest react-native CLI which does not have a `-v` option (anymore).

```

Adapter node: v10.11.0 x64
vscode-chrome-debug-core: 6.7.46
Could not debug. Seems to be that you are trying to debug from within directory that is not a React Native project root. 
 If so, please, follow these instructions: https://github.com/Microsoft/vscode-react-native/blob/master/doc/customization.md#project-structure. (error code 604)
******** Unhandled error in debug adapter - Unhandled promise rejection: Error: Error while executing command 'react-native -v' (error code 101)
    at Function.getInternalError (/Users/cknoetschke/.vscode/extensions/msjsdiag.vscode-react-native-0.10.2/src/common/error/errorHelper.js:10:16)
    at ChildProcess.spawnedProcess.once (/Users/cknoetschke/.vscode/extensions/msjsdiag.vscode-react-native-0.10.2/src/common/node/childProcess.js:44:58)
    at Object.onceWrapper (events.js:273:13)
    at ChildProcess.emit (events.js:182:13)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:240:12)
```

**Comment by SounD120** [comment]
Hi @ck86 . Do you mean the latest https://www.npmjs.com/package/@react-native-community/cli package?

**Comment by ck86** [comment]
Yes. Sorry, forgot that there was another one. :)

**Comment by SounD120** [comment]
Hey @batchu , @cvharris , @Saraceni . Which React Native CLI version do you use?
Extension developed to work with the https://www.npmjs.com/package/react-native-cli, 
https://www.npmjs.com/package/@react-native-community/cli may work but not yet supported.

**Comment by ck86** [comment]
This will solve the issue (for me). :)

**Comment by Saraceni** [comment]
@SounD120 I'm currently using

react-native-cli: 2.0.1
react-native: 0.60.4

**Comment by SounD120** [comment]
@batchu Could you please share a minimal demo application where this behavior is reproduces?

**Comment by SounD120** [comment]
@batchu Do you have a chance to look at this?

**Comment by batchu** [comment]
@SounD120  I get the following error when I try that

`OS: win32 x64
Adapter node: v10.11.0 x64
vscode-chrome-debug-core: 6.7.46
Could not debug. Cannot read property 'getCurrentUserAsync' of undefined`

I have pushed my source code [here ](https://github.com/batchu/react-native-detect-orientation)to aid with the troubleshooting process. I've committed the .vscode and .expo folder for the same.

**Comment by SounD120** [comment]
Hi @batchu . This error is a [known issue of expo-cli](https://github.com/expo/expo-cli/issues/864). 
You can use the [following instruction](https://github.com/microsoft/vscode-react-native/issues/1060#issuecomment-513836253) to resolve your issue. 
After applying a fix above I was able to debug your application with Expo.
Please let me know if you have any questions.

**Comment by SounD120** [comment]
@Saraceni Are you still experiencing this issue?

**Comment by batchu** [comment]
Unfortunately yes. Let me know if you need any information from me to troubleshoot this. Thank you.

**Comment by SounD120** [comment]
@batchu Have you tried applying the [fix](https://github.com/microsoft/vscode-react-native/issues/1060#issuecomment-513836253) described above?
To workaround your error simply reinstall `@expo/xdl` package in extension directory. Here is the exact steps:
* Open extension directory in terminal: `%USERPROFILE%\.vscode\extensions\msjsdiag.vscode-react-native-0.10.2`  
* Uninstall @expo/xdl: `npm uni --no-save @expo/xdl`
* Install @expo/xdl@56.0.0: `npm i --no-save @expo/xdl@56.0.0`

Also please make sure that node modules were installed inside your project, if not then run `npm install` and reload VS Code window inside the project.
Please let us know if it works for you.

**Comment by hedgepigdaniel** [comment]
I experienced this problem - it would be good if a useful error message was logged in case the `react-native run-<platform>` command didn't work. Currently it seems that any such error causes the debugger to stop silently without logging anything.

**Comment by SounD120** [comment]
Hi @batchu . Do you have a chance to try my recommendations? 

Hi @hedgepigdaniel in your case all output channels (View -> Toggle Output -> React-Native) was empty?

**Comment by SounD120** [comment]
Hi guys. I'm closing this issue due to inactivity. Feel free to open new issues if needed.

**Comment by cvharris** [comment]
I got this again (I get it every few days) recently and tried like I always do to **Reload Window** but this time that didn't work. I tried a bunch of things like upgrading Node, reinstalling `react-native-cli`, etc. but eventually it was literally closing VSCode and rebooting it again that kicked this back into gear.

Seems like something happens whenever VSCode is left open but my Mac has been left idle for sometime, or if the debugging session is left running. Not sure, but still an annoying bug with not a clear path for solving.

**Comment by SounD120** [comment]
Hi @cvharris . Could you please create a new issue for your specific case?

**Comment by orefors** [comment]
> 
> 
> I got this again (I get it every few days) recently and tried like I always do to **Reload Window** but this time that didn't work. I tried a bunch of things like upgrading Node, reinstalling `react-native-cli`, etc. but eventually it was literally closing VSCode and rebooting it again that kicked this back into gear.
> 
> Seems like something happens whenever VSCode is left open but my Mac has been left idle for sometime, or if the debugging session is left running. Not sure, but still an annoying bug with not a clear path for solving.

This is probably due to the React Native Tools extension being updated. Using the hack/workaround mentioned above, it will have to be applied after every update to the extension. 
I am very much looking forward to a permanent solution to this problem.

**Comment by SounD120** [comment]
Hi @orefors . When this [bug](https://github.com/expo/expo-cli/issues/864) will be fixed this problem will be resolved.

**Comment by Properko** [comment]
I'm getting the same behavior, however, the cause was using multi-root workspace in vscode.
Once I have opened the project as a folder directly, debug starts successfully.
Even though, I'm pretty sure the cwd property was correctly set (to '${workspaceFolder}') in the launch configuration.

**Comment by superandrew** [comment]
@SounD120 please check the linked expo cli issue for resolution 👍

**Comment by SounD120** [comment]
Thanks @superandrew . Fixed in #1155
